etc/init.d/zfs-zed.in

```diff
@@ -45,8 +39,7 @@ ZED_PIDFILE="@runstatedir@/$ZED_NAME.pid"
 
 do_depend()
 {
-	before zfs-import
-	after sysfs
+	after sysfs zfs-mount
```

Comment:
That's sort of the whole point of this change.  zfs-mount mounts `/var` when `/var` is a separate dataset.  ZED writes `/var/lib/run/zed.state` so `/var` needs to be mounted first.